### ISSUE TITLE
Helm chart complains about missing value if using a pulumi.Output from the Random plugin

### ISSUE THREAD

**Issue by mattfysh** [pulumi/pulumi-kubernetes#3260]
### What happened?

the code below produces the following error when running `pulumi up`

```
    error: kubernetes:helm.sh/v4:Chart resource 'dagster' has a problem: values don't meet the specifications of the schema(s) in the following chart(s):
    dagster:
    - postgresql: postgresqlPassword is required
```

### Example

```ts
import * as pulumi from '@pulumi/pulumi'
import * as k8s from '@pulumi/kubernetes'
import * as random from '@pulumi/random'

const psql = (() => {
  const host = `dagster-postgresql`
  const user = 'dagstersu'
  const rnd = new random.RandomBytes('sasl-password', { length: 32 })
  const pass = pulumi.interpolate`dagster_${rnd.hex}`
  const db = 'dagster'
  return { host, user, pass, db }
})()

new k8s.helm.v4.Chart(
  'dagster',
  {
    repositoryOpts: {
      repo: 'https://dagster-io.github.io/helm',
    },
    chart: 'dagster',
    values: {
      postgresql: {
        postgresqlHost: psql.host,
        postgresqlUsername: psql.user,
        postgresqlPassword: pulumi.output(psql.pass),
        postgresqlDatabase: psql.db,
      },
    }
  },
  {
    transforms: [
      args => args
    ]
  }
)

```

### Output of `pulumi about`

CLI          
Version      3.136.1
Go Version   go1.23.2
Go Compiler  gc

Plugins
KIND      NAME        VERSION
resource  kubernetes  4.18.1
language  nodejs      3.136.1-dev.0
resource  random      4.16.6

Host     
OS       darwin
Version  14.6.1
Arch     arm64

This project is written in nodejs: executable='/opt/homebrew/bin/node' version='v22.9.0'

Current Stack: mattfysh/bug/dev

TYPE                 URN
pulumi:pulumi:Stack  urn:pulumi:dev::bug::pulumi:pulumi:Stack::bug-dev

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mattfysh** [comment]
im not sure were the bug is coming from, lots of different things can make the bug go away, e.g.

1. removing the transform function (which I need for other reasons)
2. using a static string for `postgresqlPassword`
3. or, instead of static string, any other `pulumi.output` value that is being produced by the Random plugin

**Comment by blampe** [comment]
@mattfysh does this only occur during preview? E.g. does `pulumi up --skip-preview` succeed?

The provider is calling the equivalent of `helm template`, and this chart has a schema (which is great) but during preview the password isn't known yet.

> or, instead of static string, any other pulumi.output value that is not being produced by the Random plugin

Are you wrapping static strings with `pulumi.output` or generating something dynamically with a different provider? I suspect the problem comes down to whether the output is resolvable at preview time.

**Comment by mattfysh** [comment]
> does pulumi up --skip-preview succeed

yes, it does seem to work when using `--skip-preview` ie. the helm chart was successfully deployed

> Are you wrapping static strings with pulumi.output or generating something dynamically with a different provider?

I tried a couple of things e.g.

```
postgresqlPassword: 'static-string'
postgresqlPassword: pulumi.output('test')
```

You mention "whether the output is resolvable at preview time" but what confuses me is why this hinges on the presence of the `transforms` function? When I remove it, everything works again - how would the presence of a transform affect which outputs can be resolved at the time of preview & schema validation? Instead, when there is no transform function, I get the following warning during preview (but I am still able to proceed with deployment):

```
  kubernetes:helm.sh/v4:Chart (dagster):
    warning: Input properties have unknown values. Preview is incomplete.
```

**Comment by mattfysh** [comment]
fyi, I have had issues with the random plugin in the past - im not sure where this bug is coming from, but for some context: https://github.com/pulumi/pulumi-random/issues/607

**Comment by blampe** [comment]
> how would the presence of a transform affect which outputs can be resolved at the time of preview & schema validation? Instead, when there is no transform function, I get the following warning during preview (but I am still able to proceed with deployment)

That's a very good question, I didn't realize the behavior depends on the presence of `transforms` but that "Preview is incomplete" warning is definitely what I would _expect_ to see regardless of `transforms`. We'll need to look into this.

**Comment by rquitales** [comment]
I discovered a workaround for this, I believe the error doesn't occur if we use a global `registerResourceTransform` instead of defining the transform inline with the resource. For example:

```typescript
...
new k8s.helm.v4.Chart(
  'dagster',
  {
    repositoryOpts: {
      repo: 'https://dagster-io.github.io/helm',
    },
    chart: 'dagster',
    values: {
      postgresql: {
        postgresqlHost: psql.host,
        postgresqlUsername: psql.user,
        postgresqlPassword: psql.pass,
        postgresqlDatabase: psql.db,
      },
    }
  },
)

// Change with whatever transform you want to do
pulumi.runtime.registerResourceTransform(async (args) => {
  if (args.type == "kubernetes:helm.sh/v4:Chart" && args.name == "dagster") {
    if (args.props.values) {
      args.props.values.postgresql.postgresqlUsername = "username_override";
    }
    return args;
  }
  return;
});
```

It seems like properties are no longer treated as an `unknown` when using the inline resource transformation. This might be a bug in upstream Pulumi, rather than a bug in the Kubernetes provider.
